packages/near-api-js/src/account.ts

```diff
@@ -687,4 +485,12 @@ export class Account {
             total: summary.total.toString(),
         };
     }
+
+    createTransaction(receiver: Account | string): TransactionBuilder {
+        return new TransactionBuilder(this.connection, this.accountId, typeof receiver === 'string' ? receiver : receiver.accountId);
```

Comment:
Rather than refactor the `Account` class to share some functionality, why not pass the `signTransaction` and `signAndSendTransaction` methods to the new class? `TransactionBuilder` doesn't need to know how to send transactions, as evidenced by the name 🙂 

Passing the methods would eliminate the need to break up the `Account` class, which seems to have been done solely for the purpose of factoring out two methods (and their dependencies). What would you think about something like this instead?

```ts
createTransaction(receiverId: string): TransactionBuilder {
  return new TransactionBuilder({
    signTransaction: (actions: Action[]) => this.signTransaction(this.accountId, actions),
    signAndSendTransaction: (actions: Action[]) => this.signAndSendTransactions({ receiverId: this.accountId, actions }),
  });
}
...
class TransactionBuilder {
  signTransaction: (actions: Action[]) => Promise<[Uint8Array, SignedTransaction]>;
  signAndSendTransaction: (actions: Action[]) => Promise<FinalExecutionOutcome>;

  constructor({ signTransaction, signAndSendTransaction }) {
    this.signTransaction = signTransaction;
    this.signAndSendTransaction = signAndSendTransaction;
  }
  ...
}
```

Now `TransactionBuilder` doesn't need a `connection` or `accountId` - it just builds transactions and exposes `sign` and `signAndSend` methods that look like
```ts
sign(): Promise<[Uint8Array, SignedTransaction]> {
  return this.signTransaction(this.actions);
}
```

With this the refactor is not required and we can keep `Account` logic consolidated.

Reply:
I see what you mean about its use outside of `Account`, thanks for the clarification.

> Separating out how `Account` sends transactions from `Account` specific functionality has been overdue imo.

I think you're 100% right on this, however I would argue that factoring this behavior out into a parent class doesn't go far enough in separating these concerns. Inheritance may be more convenient in the short term for this but we have an opportunity to use composition to accomplish the same thing in a more robust way. By passing instances of `TransactionSender` to `Account` and `TransactionBuilder` rather than putting this in a parent class, we can avoid the coupling that inheritance guarantees.

Another benefit is that mocking becomes trivial - different `TransactionSender` mocks (e.g. one for throwing an error, another returning unexpected data, etc.) could be shared across unit tests, eliminating the need for transaction signing integration tests on consumers of `TransactionBuilder`.

Reply:
Great points. I rebased this PR on #1018 and #1019 👍